Client/WebFilters/CosmeticFiltersResourceDownloader.swift

```diff
@@ -41,23 +42,39 @@ class CosmeticFiltersResourceDownloader {
         if now.timeIntervalSince(lastFetchDate) >= fetchInterval {
             lastFetchDate = now
             
-            // MUST re-create the engine otherwise we get insane load times when calling:
-            // `engine_add_resources`
-            // This is because `engine_add_resources` will ADD resources, and not delete old ones
-            // Thus we get a huge amount of memory usage and slow down.
-            engine = AdblockRustEngine()
-            
-            loadDownloadedFiles()
-            downloadCosmeticSamples()
-            downloadResourceSamples()
+            Task {
+                // All operations must be done on a new engine,
+                // otherwise we get insane load times when calling:
+                // `engine_add_resources` on an existing engine
+                // This is because `engine_add_resources` will ADD resources, and not delete old ones
+                // Thus we get a huge amount of memory usage and slow down.
+                let newEngine = AdblockRustEngine()
+                
+                // We only setup files on the new engine to avoid race-conditions
+                do {
+                    try await loadDownloadedFiles(into: newEngine)
+                } catch {
+                    log.error("Error Loading Cosmetic-Filters: \(error)")
+                }
+                
+                do {
+                    try await downloadCosmeticSamples(with: newEngine)
+                    try await downloadResourceSamples(with: newEngine)
```

Comment:
These previously ran async, they now run sequentially, is that intended?

Reply:
Fixed.